packages/melos/lib/src/commands/run.dart

```diff
@@ -275,48 +275,24 @@ mixin _RunMixin on _Melos {
     Script script,
     Map<String, String> environment,
   ) async {
-    for (final step in steps) {
-      final scriptCommand = _buildScriptCommand(step, scripts);
-
-      final scriptSourceCode = targetStyle(
-        step.withoutTrailing('\n'),
-      );
-
-      await _executeAndLogCommand(
-        script,
-        scriptSourceCode,
-        scriptCommand,
-        environment,
-      );
-    }
-  }
-
-  Future<void> _executeAndLogCommand(
-    Script script,
-    String scriptSourceCode,
-    String scriptCommand,
-    Map<String, String> environment,
-  ) async {
-    logger.command('melos run ${script.name}');
-    logger.child(scriptSourceCode).child(runningLabel).newLine();
-
-    final exitCode = await startCommand(
-      [scriptCommand],
+    final shell = PersistentShell(
       logger: logger,
-      environment: environment,
```

Comment:
We need to set the environment in the new shell right? So that it isn't missing set environment variables etc, right?

Reply:
Right, I just added it as a parameter to initialize the persistent shell